wcfsetup/install/files/ts/WoltLabSuite/Core/I18n/Plural.ts

```diff
@@ -9,184 +9,105 @@
 
 import * as StringUtil from "../StringUtil";
 
-const PLURAL_FEW = "few";
-const PLURAL_MANY = "many";
-const PLURAL_ONE = "one";
-const PLURAL_OTHER = "other";
-const PLURAL_TWO = "two";
-const PLURAL_ZERO = "zero";
-
-const Plural = {
-  /**
-   * Returns the plural category for the given value.
-   */
-  getCategory(value: number, languageCode?: string): string {
-    if (!languageCode) {
-      languageCode = document.documentElement.lang;
-    }
-
-    // Fallback: handle unknown languages as English
-    if (typeof Plural[languageCode] !== "function") {
-      languageCode = "en";
-    }
-
-    const category = Plural[languageCode](value);
-    if (category) {
-      return category;
-    }
-
-    return PLURAL_OTHER;
-  },
-
-  /**
-   * Returns the value for a `plural` element used in the template.
-   *
-   * @see    wcf\system\template\plugin\PluralFunctionTemplatePlugin::execute()
-   */
-  getCategoryFromTemplateParameters(parameters: object): string {
-    if (!parameters["value"]) {
-      throw new Error("Missing parameter value");
-    }
-    if (!parameters["other"]) {
-      throw new Error("Missing parameter other");
-    }
-
-    let value = parameters["value"];
-    if (Array.isArray(value)) {
-      value = value.length;
-    }
-
-    // handle numeric attributes
-    const numericAttribute = Object.keys(parameters).find((key) => {
-      return key.toString() === (~~key).toString() && key == value;
-    });
-
-    if (numericAttribute) {
-      return numericAttribute;
-    }
-
-    let category = Plural.getCategory(value);
-    if (!parameters[category]) {
-      category = PLURAL_OTHER;
-    }
-
-    const string = parameters[category];
-    if (string.indexOf("#") !== -1) {
-      return string.replace("#", StringUtil.formatNumeric(value));
-    }
-
-    return string;
-  },
-
-  /**
-   * `f` is the fractional number as a whole number (1.234 yields 234)
-   */
-  getF(n: number): number {
-    const tmp = n.toString();
-    const pos = tmp.indexOf(".");
-    if (pos === -1) {
-      return 0;
-    }
-
-    return parseInt(tmp.substr(pos + 1), 10);
-  },
-
-  /**
-   * `v` represents the number of digits of the fractional part (1.234 yields 3)
-   */
-  getV(n: number): number {
-    return n.toString().replace(/^[^.]*\.?/, "").length;
-  },
-
+enum Category {
```

Comment:
```suggestion
const enum Category {
```
This allows the compiler to inline the value and remove the enum from the final code. There is no partical use of carrying it around on runtime?

Reply:
I attempted that, but somehow that caused issues with defining the `Parameters` type. I've re-tested this and apparently my later changes fixed that definition issue.